crates/core/src/agent/state.rs

```diff
@@ -159,13 +161,29 @@ impl AddressableContent for AgentStateSnapshot {
 // @TODO abstract this to a standard trait
```

Comment:
If this really closes #196 then can remove this comment?

Reply:
Yeah I could use some background on this, I never even knew we had ActionResponses. Sounds GhostEndpoint-y